### DISCUSSION TITLE
Incorrect plotting of dxf files

### DISCUSSION THREAD

**Discussion by stansmolders** [mozman/ezdxf#1178] [Q&A] [answered]
Hi, I am trying to plot simple dxf files with a variety of shapes. For some reason (when dxf files contain the entities arcs or circles), some entities seem to be plotted wrong. For some dxf files this behaviour seems to be quite random as e.g. in the same plot, some arcs can be plotted correctly on the positive plane, while others seem to be mirrored (x=-x). Adding x=-x however would mess up the correctly plotted entities, so this would not be a solution. Also, when plotting some dxf files, there is no problem at all. The problem seems to occur due to the use of dxf.center. Is this a known problem or am I using it wrong? I added screenshots of this behaviour below and the code i use for this part is as follows:

           def _plot_dxf(self, doc):
        """Plot the DXF file using matplotlib."""
        print("Plotting entities from the DXF file...")
        msp = doc.modelspace()
        fig, ax = plt.subplots()

        for entity in msp:
            print(f"Processing entity: {entity.dxftype()}")
            if entity.dxftype() == 'LINE':
                start = entity.dxf.start
                end = entity.dxf.end
                print(f"Plotting LINE from {start} to {end}")
                ax.plot([start.x, end.x], [start.y, end.y], 'b')
            elif entity.dxftype() == 'CIRCLE':
                center = entity.dxf.center
                radius = entity.dxf.radius
                print(f"Plotting CIRCLE at {center} with radius {radius}")
                circle = plt.Circle((center.x, center.y), radius, color='r', fill=False)
                ax.add_artist(circle)
            elif entity.dxftype() == 'ARC':
                center = entity.dxf.center
                radius = entity.dxf.radius
                start_angle = entity.dxf.start_angle
                end_angle = entity.dxf.end_angle
                print(f"Plotting ARC at {center} with radius {radius}, from {start_angle}° to {end_angle}°")
                arc = Arc((center.x, center.y), radius*2, radius*2, angle=0, theta1=start_angle, theta2=end_angle, color='g')
                ax.add_patch(arc)
            elif entity.dxftype() == 'ELLIPSE':
                center = entity.dxf.center
                major_axis = entity.dxf.major_axis
                print(f"Plotting ELLIPSE at {center} with major axis {major_axis}")
                ellipse = plt.Circle((center.x, center.y), major_axis.magnitude, color='y', fill=False)
                ax.add_artist(ellipse)
        ax.set_aspect('equal', adjustable='datalim')
        plt.title("DXF Viewer")
        plt.xlabel("X-axis")
        plt.ylabel("Y-axis")
        plt.grid(True)
        print("Displaying plot...")
        plt.show()

![dxf plot4](https://github.com/user-attachments/assets/9d88fcd8-197a-45be-91b6-f94c828680f8)
![dxf plot3](https://github.com/user-attachments/assets/0453ffa8-0ea7-4c2f-ab14-2bc19b26a759)
![dxf plot2](https://github.com/user-attachments/assets/ca624db8-19aa-4a57-a6d6-b8278ac04d74)
![dxf plot](https://github.com/user-attachments/assets/bfb73efc-a4ec-4f73-9670-2acb83babc9e)

**Comment by mbway** [comment]
Are you aware of the ezdxf drawing add-on? Have you compared your implementation with that to see where the logic differs?

**Comment by stansmolders** [comment]
> Are you aware of the ezdxf drawing add-on? Have you compared your implementation with that to see where the logic differs?

Could you provide a link to what you mean? I basically only need to plot dxf files correctly for this part, so if this already available somewhere, other logic would be fine as well.

**Reply by mbway** [reply]
https://ezdxf.readthedocs.io/en/stable/addons/drawing.html

The add-on can take a DXF and output PNG, s g, pdf and show an interactive Qt GUI

**Comment by mozman** [comment, marked as answer]
You have to take the Object Coordinate System (OCS) into account. 
Please read this: https://ezdxf.mozman.at/docs/concepts/ocs.html

**Reply by stansmolders** [reply]
This worked, thanks
